### PR TITLE
feature: Add IQM Garnet device, retire OQC Lucy device

### PR DESCRIPTION
**Pull request by rmshaffer**
*Issue #, if available:*

*Description of changes:*
Mark OQC Lucy device with `_` prefix since it is now retired from the Braket service. Add IQM Garnet device which recently launched.

*Testing done:*
`julia -e 'using Pkg; Pkg.test("Braket")'`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/awslabs/braket-jl/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/awslabs/braket-jl/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/awslabs/braket-jl/blob/main/README.md) and [API docs](https://github.com/awslabs/braket-jl/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have installed and run [`git secrets`](https://github.com/awslabs/git-secrets) to make sure I did not commit any sensitive information (passwords or credentials)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/integ_tests/device_creation.jl [hunk 1]
```diff
@@ -5,9 +5,10 @@ RIGETTI_ARN = "arn:aws:braket:::device/qpu/rigetti/Aspen-11"
 IONQ_ARN = "arn:aws:braket:us-east-1::device/qpu/ionq/Aria-1"
 SIMULATOR_ARN = "arn:aws:braket:::device/quantum-simulator/amazon/sv1"
 OQC_ARN = "arn:aws:braket:eu-west-2::device/qpu/oqc/Lucy"
```
Comment (kshyatt-aws) [inline test/integ_tests/device_creation.jl:7]: Should we remove `OQC_ARN` from these tests?

Comment (rmshaffer) [reply test/integ_tests/device_creation.jl:7]: I mean, `DWAVE_ARN` is still here. 🤷 Up to you, happy to remove it.

File: test/schemas_misc.jl [hunk 2]
```diff
@@ -209,7 +209,7 @@ using Braket, Braket.IR, Test, JSON3, StructTypes
     end
 
     @testset "Device metadata" begin
-        @testset for T in (Braket.OqcMetadata, Braket.XanaduMetadata, Braket.RigettiMetadata)
+        @testset for T in (Braket.OqcMetadata, Braket.IqmMetadata, Braket.XanaduMetadata, Braket.RigettiMetadata)
```
Comment (kshyatt-aws) [inline test/schemas_misc.jl:212]: Should we drop `OqcMetadata` here?

**Review by kshyatt-aws — APPROVED** [review]
Might want to add some docs for the new device too but otherwise LGTM

File: test/schemas_misc.jl (hunk 2 above)
Comment (rmshaffer) [reply test/schemas_misc.jl:212]: Maybe? `XanaduMetadata` is still here. I'd be happy to make a pass through and remove all of the retired devices.

Comment (kshyatt-aws) [reply test/schemas_misc.jl:212]: I guess I'm wondering if we're still testing these on the Python side

Comment (rmshaffer) [reply test/schemas_misc.jl:212]: Good point. No, we're not. The code for the retired devices is still there for backward compatibility, but the tests have been removed (example: https://github.com/amazon-braket/amazon-braket-sdk-python/pull/1014). I can make the same change here if it makes sense to you.

**Comment by rmshaffer** [comment]
Waiting for #92 to be merged to fix the PyBraket-unit CI issues.

**Pull request merged**